integration/users/usersinfo_test.go

```diff
@@ -0,0 +1,496 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package users
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/FerretDB/FerretDB/integration"
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+// createUser creates a bson.D command payload to create an user with the given username and password.
+func createUser(username, password string) bson.D {
+	return bson.D{
+		{"createUser", username},
+		{"roles", bson.A{}},
+		{"pwd", password},
+	}
+}
+
+func TestUsersinfo(t *testing.T) {
+	t.Parallel()
+
+	ctx, collection := setup.Setup(t)
+	client := collection.Database().Client()
+
+	// Map of users per database suffix
+	dbToUsers := []struct {
+		dbSuffix string
+		payloads []bson.D
+	}{
+		{
+			dbSuffix: "",
+			payloads: []bson.D{
+				createUser("one", "pwd1"),
+				createUser("two", "pwd2"),
+			},
+		},
+		{
+			dbSuffix: "_example",
+			payloads: []bson.D{
+				createUser("a", "password1"),
+				createUser("b", "password2"),
+				createUser("c", "password3"),
+			},
+		},
+		{
+			dbSuffix: "_few",
+			payloads: []bson.D{
+				createUser("i", "password1"),
+				createUser("j", "password2"),
+			},
+		},
+		{
+			dbSuffix: "_another",
+			payloads: []bson.D{
+				createUser("singleuser", "123456"),
+			},
+		},
+	}
+
+	dbPrefix := testutil.DatabaseName(t)
+
+	// Create users into the database to test userInfo.
+	for _, inserted := range dbToUsers {
+		dbName := testutil.DatabaseName(t) + inserted.dbSuffix
+		db := client.Database(dbName)
+
+		// Clear any residual database users before recreating them.
```

Comment:
I don't understand how any residual database users could remain. How that's possible? Why do we need to call `dropAllUsersFromDatabase` manually?

Reply:
You're correct. This shouldn't happen. I had wrongly assumed that a test failure that wasn't cleaned via `t.Cleanup()` might leave data on the database early on when creating the tests, but this doesn't apply here as we store all user information in the admin database. Removing the code.